components/src/core/components/Calendar/Calendar.vue

```diff
@@ -0,0 +1,147 @@
+<script lang="ts">
+import {
+  parseDate,
+  format,
+  isEqual,
+  getDaysInMonth,
+  getYear,
+  getMonth,
+  getDayOffset,
+  freshDate,
+} from '../../../utils/date';
+import {computed, defineComponent, h, PropType, reactive, toRefs} from 'vue';
+import {DateOptions, DAYS_OF_WEEK, MONTHS_OF_YEAR, SUNDAY} from './types';
+import Day from '@orangehrm/oxd/core/components/Calendar/Day.vue';
+import DateVue from '@orangehrm/oxd/core/components/Calendar/Date.vue';
+import CalendarController from '@orangehrm/oxd/core/components/Calendar/CalendarController.vue';
+
+export default defineComponent({
+  name: 'oxd-calendar',
+  props: {
+    modelValue: {
+      type: [String, Date],
+      default: () => {
+        return freshDate();
+      },
+    },
+    format: {
+      type: String,
+      default: 'yyyy-MM-dd',
+    },
+    firstDayOfWeek: {
+      type: Number,
+      default: 0, // 0 | 1 | 2 | 3 | 4 | 5 | 6 => 0 represents Sunday
+    },
+    years: {
+      type: Array,
+      default: () => {
+        return Array.from(
+          {length: getYear(new Date()) - 1969},
+          (_, i) => 1970 + i,
+        );
+      },
+    },
+    months: {
+      type: Array,
+      default: () => MONTHS_OF_YEAR,
```

Comment:
is it good load defaults by calendar local settings, 
I mean if local en_EN then load strings via en file
@RajithaKumara what do you think?

Reply:
@samanthajayasinghe ah i thought about this. `date-fn` has locales from where we can get names of months/days in different locales rather than passing as props. but in such scenario ability to give custom names to months/days shouldn't be possible.

currently our api is like => `<oxd-calendar :months="monthslist" :days="dayslist"/>
`
revised API would be => `<oxd-calendar locale="en-us"/>`

Reply:
I think we can keep `:locale` props as well as `:months` and :days props but `:locale` is superset of `months` and `daylist`

Reply:
`locale` props added!